plugins/dell/fu-plugin-dell.c

```diff
@@ -995,6 +997,11 @@ fu_dell_toggle_flash (FuPlugin *plugin, FuDevice *device,
 			g_debug ("Toggled MST hub GPIO to %d", enable);
 	}
 
+#if defined (HAVE_SYNAPTICS)
+	/* set a delay to allow OS response to settling the GPIO change */
+	if (enable && device == NULL && (has_dock || has_host))
```

Comment:
is `device == NULL` right?

Reply:
It's only supposed to run in the coldplug path and the plugin explicitly sets device to NULL in that path.